wordpress-mobile/WordPress-iOS
# [Notifications P2] Commen Moderation: Pending

Fixes #https://github.com/Automattic/wordpress-mobile/issues/33

## Description
This PR adds the new Moderation View UI to the Comment Detail Screen. The functionality will follow with other PRs.

## Acceptance Criteria
If the comment has the "Pending" moderation status, a moderation panel is displayed at the bottom of the comment details screen. There is a separator between the content and the panel.
1‍⃣ There is a label indicating the moderation state with a dedicated icon.
2‍⃣ Approve comment CTA button with an icon is visible but not actionable.
3‍⃣ More options CTA button opens a bottom sheet dialog containing all the available moderation options.
4‍⃣ The bottom sheet dialog presents "Approved," "Pending," "Thrash," and "Spam" options that are not actionable.

## Testing Steps
1.  Install and Login on Jetpack App
2. Go to Notifications
3. Select a comment notification that has "pending" state (the moderation can be set from web if no comment is in pending state.
4. ✅  Expect the new moderation component to appear instead of the old one.
5. ✅ Verify if tapping on "More options" show the moderation options sheet.

_Test in dark and light mode and with different dynamic font sizes._

## Screenshot Reference

| Header | Header |
|--------|--------|
| ![Simulator Screenshot - iPhone 15 - 2024-05-09 at 20 03 32](https://github.com/wordpress-mobile/WordPress-iOS/assets/15968946/4fa1ef96-be65-4f18-b9fc-d998b85b9b94) | ![Simulator Screenshot - iPhone 15 - 2024-05-09 at 20 02 45](https://github.com/wordpress-mobile/WordPress-iOS/assets/15968946/b610108d-9973-4c90-a336-7f52b77d599b) | 

## Regression Notes
1. Potential unintended areas of impact


7. What I did to test those areas of impact (or what existing automated tests I relied on)


8. What automated tests I added (or what prevented me from doing so)

PR submission checklist:


Testing checklist:
